Sources/Purchasing/PurchasesDelegate.swift

```diff
@@ -47,13 +47,24 @@ import Foundation
     /**
      * Called when a user initiates a promotional in-app purchase from the App Store.
      * If your app is able to handle a purchase at the current time, run the deferment block in this method.
-     * If the app is not in a state to make a purchase: cache the defermentBlock,
-     * then call the defermentBlock when the app is ready to make the promotional purchase.
-     * If the purchase should never be made, you don't need to ever call the defermentBlock and
-     * ``Purchases`` will not proceed with promotional purchases.
-
+     * If the app is not in a state to make a purchase: cache the `startPurchase` block,
+     * then call the `startPurchase` block when the app is ready to make the promotional purchase.
+     *
+     * If the purchase should never be made, you don't need to ever call the block and
+     * ``Purchases`` will not proceed with the promotional purchase.
+     *
      * - Parameter product: `StoreProduct` the product that was selected from the app store
      */
+    @objc optional func purchases(_ purchases: Purchases,
+                                  isReadyForPromotedProduct product: StoreProduct,
+                                  purchase startPurchase: @escaping DeferredPromotionalPurchaseBlock)
```

Comment:
The main change of this PR. I changed this slightly from @aboedo's suggestion, I think `purchases:isReadyForPromotedProduct:purchase:` reads nicely.

Reply:
i was finding the is/has prefix a little confusing, considering there is no boolean involved here. is that a swifty and/or delegate thing? 

maybe something without is/has like `receievedPromotedProductPurchaseHook` or something. i don't like that necessarily just spitballing...

Reply:
> i was finding the is/has/should prefix a little confusing, considering there is no boolean involved here. is that a swifty and/or delegate thing?

Definitely more common for a Boolean, but I _think_ this is also a Cocoa-like name for a protocol method?